test/remote/gateways/remote_barclaycard_smartpay_test.rb

```diff
@@ -152,6 +153,16 @@ def test_successful_purchase_with_no_address
     assert_equal '[capture-received]', response.message
   end
 
+  def test_successful_authorize_with_threeds
+    assert response = @gateway.authorize(@amount, @three_ds_enrolled_card, @options.merge(execute_threed: true))
+    assert_equal 'RedirectShopper', response.message
+    assert response.test?
+    assert !response.authorization.blank?
```

Comment:
There's `refute` and friends instead of `assert !...` if you want it here, but I'm fine with this pattern also.